tests/x11/tomcat.pm

```diff
@@ -62,6 +62,119 @@ sub run() {
     my $websocket_test = Tomcat::WebSocketsTest->new();
     $websocket_test->test_all_examples();
 
+    # Close firefox
+    $self->close_firefox();
+    assert_screen('generic-desktop');
+
+    # Install and configure apache2 and apache2-mod_jk connector
+    $self->select_serial_terminal();
+    record_info('install and configure apache2 and apache2-mod_jk connector Setup');
+    zypper_call('in apache2 apache2-mod_jk');
+
+    # Connection from apache2 to tomcat: tomcat part
+    systemctl('stop tomcat');
+
+    # Define an AJP 1.3 Connector on IPv6 localhost port 8009
+    my $ajp_connector = qq(    <Connector protocol="AJP/1.3" address="::1" port="8009" redirectPort="8443" \/>);
+    assert_script_run qq(sed -i '/<\!-- Define an AJP 1.3 Connector on port 8009 -->/ a $ajp_connector\' /etc/tomcat/server.xml);
+
+    # Define a worker "ajp13" which listens behind the above AJP 1.3 Connector address and port
+    assert_script_run('cp /usr/share/doc/packages/apache2-mod_jk/workers.properties /etc/tomcat/workers.properties');
+    assert_script_run(
+        "echo \"\$(cat <<EOF
+worker.list=ajp13
+worker.ajp13.reference=worker.template
+worker.ajp13.host=localhost
+worker.ajp13.port=8009
+EOF
+	)\" >> /etc/tomcat/workers.properties", timeout => 180
+    );
+    # Connection from apache2 to tomcat: apache2 part
+    systemctl('stop apache2');
+    assert_script_run('cp -ai /usr/share/doc/packages/apache2-mod_jk/jk.conf /etc/apache2/conf.d');
+    if (is_sle('=15') or is_sle('<12-sp4')) {
+        record_soft_failure 'boo#1167896';
+        assert_script_run(
+            "echo \"\$(cat <<EOF
+JkShmFile /var/log/tomcat/jk-runtime-status
+EOF
+	)\" >> /etc/apache2/conf.d/jk.conf", timeout => 180
+        );
+        assert_script_run("sed -i 's|servlets-examples|examples/servlets|g' /etc/apache2/conf.d/jk.conf");
+        assert_script_run("sed -i 's|jsp-examples|examples/jsp|g' /etc/apache2/conf.d/jk.conf");
+    }
+    $self->load_jk_module();
+    $self->switch_to_desktop();
+    # start firefox
+    $self->start_firefox_with_profile();
+    $self->firefox_open_url('http://localhost/examples/servlets');
+    send_key_until_needlematch('tomcat-servlet-examples-page', 'ret');
+
+    $self->firefox_open_url('http://localhost/examples/jsp');
+    send_key_until_needlematch('tomcat-jsp-examples', 'ret');
+    $self->close_firefox();
+
+    $self->select_serial_terminal();
+    systemctl('start tomcat');
+
+    $self->switch_to_desktop();
+    # start firefox
+    $self->start_firefox_with_profile();
+    record_info('Servlet Testing');
+    $servlet_test = Tomcat::ServletTest->new();
+    $servlet_test->test_all_examples(1);
```

Comment:
I would recommend doing (It's purely cosmetic)

```suggestion
    my $with_jk = 1;
    $servlet_test->test_all_examples($with_jk);
```

Reply:
Sure I have taken the point. :)